content/shmem_atomic_fetch_inc_nbi.tex

```diff
@@ -0,0 +1,55 @@
+\color{Green}
+\apisummary{
+    This nonblocking atomic routine performs an atomic fetch-and-increment
+    operation on a remote data object.
+}
+
+\begin{apidefinition}
+
+\begin{C11synopsis}
+void @\FuncDecl{shmem\_atomic\_fetch\_inc\_nbi}@(TYPE *fetch, TYPE *dest, int pe);
+void @\FuncDecl{shmem\_atomic\_fetch\_inc\_nbi}@(shmem_ctx_t ctx, TYPE *fetch, TYPE *dest, int pe);
+\end{C11synopsis}
+where \TYPE{} is one of the standard \ac{AMO} types specified by
+Table~\ref{stdamotypes}.
+
+\begin{Csynopsis}
+void @\FuncDecl{shmem\_\FuncParam{TYPENAME}\_atomic\_fetch\_inc\_nbi}@(TYPE *fetch, TYPE *dest, int pe);
+void @\FuncDecl{shmem\_ctx\_\FuncParam{TYPENAME}\_atomic\_fetch\_inc\_nbi}@(shmem_ctx_t ctx, TYPE *fetch, TYPE *dest, int pe);
+\end{Csynopsis}
+where \TYPE{} is one of the standard \ac{AMO} types and has a corresponding
+\TYPENAME{} specified by Table~\ref{stdamotypes}.
+
+\begin{apiarguments}
+
+\apiargument{IN}{ctx}{A context handle specifying the context on which to
+    perform the operation. When this argument is not provided, the
+    operation is performed on the default context.}
+\apiargument{OUT}{fetch}{Local data object to be updated.}
+\apiargument{OUT}{dest}{The remotely accessible data object to be updated on the
+    remote \ac{PE}.}
+\apiargument{IN}{pe}{An integer that indicates the \ac{PE} number on which
+    \dest{} is to be updated.}
+
+\end{apiarguments}
+
+
+\apidescription{
+   These nonblocking \FUNC{shmem\_atomic\_fetch\_inc\_nbi} routines perform an
+   atomic fetch-and-increment operation. This routine returns after posting the
+   operation. The operation is considered complete after a subsequent call to
+   \FUNC{shmem\_quiet}. At the completion of \FUNC{shmem\_quiet}, the \dest{} on
+   \ac{PE} \VAR{pe} is increased by one and the previous contents of \dest{} are
+   fetched into the \VAR{fetch} local data object as one atomic operation.
```

Comment:
Phrasing of "as one atomic operation" needs to be reworded so that the realm of atomicity doesn't extend through the write of output data into the output buffer.

Reply:
removed based on RMA WG meeting suggestion